src/math/trig.c

```diff
@@ -94,9 +108,159 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
-	/* Q2.30 format */
+	/* Q2.30 format -sine,cosine*/
 	*th_cdc_fxp = th_rad_fxp;
+}
+
+/**
+ * \CORDIC-based approximation for inverse sine, cosine
+ * Arguments	: int32_t realvalue
+ *		  int16_t numiters
+ *		  const int32_t LUT[CORDIC_30B_ITABLE_SIZE][30]
+ * Return Type	: int32_t
+ */
+int32_t is_scalar_cordic(int32_t realvalue, int16_t numiters,
```

Comment:
Same thought here. Would it be better to set as inline?

Reply:
Too large function in my opinion for inline.

Reply:
too large atm, but this should really be 2 functions. @ShriramShastry we are switching on type twice here for EN_32B_CORDIC_ASINE and EN_32B_CORDIC_ACOSINE (and in a loop too). This should be 2 functions, one for EN_32B_CORDIC_ASINE and the other for EN_32B_CORDIC_ACOSINE